plugin/Form/FormAddressCreator.php

```diff
@@ -236,4 +242,128 @@ private function getUserNameById(string $userId): string
+	{
+		if ($addressId > $latestAddressIdOnEnterPrise || is_null($latestAddressIdOnEnterPrise)) {
+			return '';
+		}
+
+		return $this->handleLogicMessageDuplicateAddressData($pFormData, $addressId);
+	}
+
+
+	/**
+	 * @return mixed
+	 * @throws ApiClientException
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 */
+	public function getLatestAddressIdInOnOfficeEnterprise()
+	{
+		$pContainerBuilder = new ContainerBuilder;
+		$pContainerBuilder->addDefinitions(ONOFFICE_DI_CONFIG_PATH);
+		$pContainer = $pContainerBuilder->build();
+		$pContainer->get(CacheHandler::class)->clear();
+
+		$requestParams = [
+			'sortby' => 'KdNr',
+			'sortorder' => 'DESC',
+			'listlimit' => 1
+		];
+		$pApiClientAction = new APIClientActionGeneric
+			($this->_pSDKWrapper, onOfficeSDK::ACTION_ID_READ, 'address');
+		$pApiClientAction->setParameters($requestParams);
+		$pApiClientAction->addRequestToQueue()->sendRequests();
+		$result = $pApiClientAction->getResultRecords();
+
+		if (empty($result)) {
+			return null;
+		}
+
+		return $result[0]['id'];
+	}
+
+	/**
+	 * @param int $addressId
+	 * @param array $params
+	 * @return array
+	 * @throws ApiClientException
+	 */
+	private function getDataAddressById(int $addressId, array $params): array
+	{
+		$requestParams = [
+			'data' => array_keys($params)
+		];
+
+		$pApiClientAction = new APIClientActionGeneric
+			($this->_pSDKWrapper, onOfficeSDK::ACTION_ID_READ, 'address');
+		$pApiClientAction->setParameters($requestParams);
+		$pApiClientAction->setResourceId((string) $addressId);
+		$pApiClientAction->addRequestToQueue()->sendRequests();
+		$result = $pApiClientAction->getResultRecords();
+
+		return $result[0]['elements'];
+	}
+
+	/**
+	 * @param array $oldData
+	 * @return string
+	 */
+	private function generateMessageDuplicateAddressData(array $oldData): string
+	{
+		$messageDuplicateAddressData = '';
+		$guidanceForHandlingSituation = '';
+		$oldAddressData = '';
+
+		if (!empty($oldData)) {
+			$messageDuplicateAddressData .= "\n\n" . __( 'Data has been duplicated:', 'onoffice-for-wp-websites' ) . "\n";
+			$messageDuplicateAddressData .= "--------------------------------------------------\n";
+			foreach ($oldData as $key => $value) {
+				$oldAddressData .= $key . ': ' . $value . "\n";
+			}
+			$guidanceForHandlingSituation .= "\n" . __( 'Duplicate detected. This data record may be a duplicate of an existing data record. Check for possible duplicates and then decide whether the data record should be updated.', 'onoffice-for-wp-websites') . "\n";
+			$guidanceForHandlingSituation .= "\n" . __( 'How to search and update duplicates in onOffice enterprise:', 'onoffice-for-wp-websites') . "\n";
+			$guidanceForHandlingSituation .=  __( 'https://de.enterprisehilfe.onoffice.com/help_entries/dubletten/?lang=en', 'onoffice-for-wp-websites');
+		}
+		$messageDuplicateAddressData .= $oldAddressData . $guidanceForHandlingSituation;
```

Comment:
Could you please check my commit for multilingual email text?
I think this could be one solution.

Reply:
Yes, I tested it and the text translation properly worked. You can watch my test video.
https://files.fm/u/9au2x5brtw#/view/nbsr35j99p